src/freertos_drivers/st/Stm32Can.cxx

```diff
@@ -103,13 +132,58 @@ Stm32Can::Stm32Can(const char *name)
     : Can(name)
     , state_(CAN_STATE_STOPPED)
 {
-    /* only one instance allowed */
-    HASSERT(instances[0] == NULL);
+    /* Get dev num (digit at the end of the devname). */
+    char dev = name[strlen(name)-1];
+    HASSERT(dev >= '0');
+    int index = dev - '0';
+#if defined(CAN3)
+    HASSERT(index < 3);
+#elif defined(CAN2)
+    HASSERT(index < 2);
+#else
+    HASSERT(index < 1);
+#endif
+    
+    /* only one instance per interface allowed */
+    HASSERT(instances[index] == NULL);
 
-    instances[0] = this;
+    instances[index] = this;
+    switch (index)
+    {
+    case 0: /* CAN1... */
```

Comment:
The style guidelines require case labels to be indented four spaces.

Reply:
Fixed.